core/src/main/scala/org/apache/spark/rdd/CoalescedRDD.scala

```diff
@@ -58,7 +58,7 @@ private[spark] case class CoalescedRDDPartition(
       val parentPreferredLocations = rdd.context.getPreferredLocs(rdd, p.index).map(_.host)
       preferredLocation.exists(parentPreferredLocations.contains)
     }
-    if (parents.size == 0) 0.0 else (loc.toDouble / parents.size.toDouble)
```

Comment:
Totally agree with you. Really sorry for bothering everyone.
However, when we do the math, I think another thing we should consider is the time that will be spent on reading the code. The code will be read again and again even by the same people. Even if it can only reduce 1 second for each code reading, that will also be a huge aggregate time saving.
For this PR, I assume this part of code won't be read that much. So it isn't worth to do it.
Thanks for the sharing.